templates/swift/Sources/Services/Realtime.swift.twig

```diff
@@ -98,36 +110,38 @@ open class Realtime : Service {
         channels: Set<String>,
         payloadType: T.Type,
         callback: @escaping (RealtimeResponseEvent) -> Void
-    ) -> RealtimeSubscription {
+    ) async throws -> RealtimeSubscription {
         subscriptionsCounter += 1
-        let counter = subscriptionsCounter
+
+        let count = subscriptionsCounter
 
         channels.forEach {
             activeChannels.insert($0)
         }
 
-        activeSubscriptions[counter] = RealtimeCallback(
+        activeSubscriptions[count] = RealtimeCallback(
             for: Set(channels),
-            and: callback
+            with: callback
         )
 
         connectSync.sync {
             subCallDepth+=1
         }
 
-        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(DEBOUNCE_MILLIS)) {
-            if (self.subCallDepth == 1) {
-                self.createSocket()
-            }
-            self.connectSync.sync {
-                self.subCallDepth-=1
-            }
+        try await Task.sleep(nanoseconds: UInt64(DEBOUNCE_NANOS))
+
+        if self.subCallDepth == 1 {
+            try await self.createSocket()
+        }
+
+        connectSync.sync {
+            self.subCallDepth -= 1
         }
 
         return RealtimeSubscription {
-            self.activeSubscriptions[counter] = nil
+            self.activeSubscriptions[count] = nil
             self.cleanUp(channels: channels)
```

Comment:
I think this part is causing the issue, when the second subscribe comes in, the channel from first subscribe is removed by the cleanup as it's not in the list of current channels.

Reply:
We might need to update the cleanup method to also check for active subscriptions or perhaps we can do the cleanup on unsubscribe/close only?

Reply:
Not sure about this because when I tested, I had breakpoints set and I saw that the 2nd create socket call had all the channels. I also didn't test closing subscriptions.

Reply:
🤔 Then I might be wrong here.

Reply:
The cleanup is only run when this `close` function is called, and the `count` and `chanels` are captured by value, so it will only remove the subscription and channels that were added for that particular call to `subscribe`